src/node_main_instance.cc

```diff
@@ -175,8 +175,10 @@ std::unique_ptr<Environment> NodeMainInstance::CreateMainEnvironment(
 
   Local<Context> context;
   if (deserialize_mode_) {
-    context =
-        Context::FromSnapshot(isolate_, kNodeContextIndex).ToLocalChecked();
+    if (!Context::FromSnapshot(isolate_, kNodeContextIndex).ToLocal(&context)) {
+      *exit_code = 1;
+      return nullptr;
```

Comment:
This change doesn't really make sense either because all call sites of `CreateMainEnvironment()` call `CHECK_NOT_NULL()` on the return value. I.e., it's still going to crash, just further from the point of the actual error.

Reply:
I'm just trying to follow good practices here. Calling `ToLocalChecked` without making an `IsEmpty` check is fatal to the process.
Many people refer to node.js codebase for sample usage of v8 APIs and we wouldn't want to give them the wrong example (that's what happened to me).

Reply:
I appreciate your reply but my primary concern for our code base is maintainability, not the education of any hypothetical observers.